src/interval.jl

```diff
@@ -171,6 +171,22 @@ function Base.last(interval::Interval{T,L,R}) where {T,L,R}
     return R !== Unbounded ? interval.last : nothing
 end
 
+function Base.min(interval::Interval{T,L,R}; precision) where {T,L,R}
+    return Base.first(interval)
+end
+
+function Base.min(interval::Interval{T,Open,R}; precision) where {T,R}
```

Comment:
I deliberately didn't set precision as anything, since I feel like having a default could cause issues with different types. 

For example, [in the issue the example](https://github.com/invenia/Intervals.jl/issues/137#issue-678556586) used `eps(..)` as the default but this would cause errors on intervals that weren't floats. 

By leaving this keyword empty it makes the caller _have_ to define the type and how precise they want the function to be.

Reply:
If it's a keyword then it's not required to be called, so what would `max(I)` be? where `I` is some interval? 
I guess it would make sense to expect an error if you don't define the precision, but when the interval is closed you don't need to supply a precision, so it seems somewhat inconsistent in that way? 

That being said I'm not really sure how to solve that issue, as you are correct that different types would require different defaults.

Reply:
max(I) would cause an error since precision wouldn't be defined. 

One workaround I can think of is to define precision as None by default in the min/max functions that are `Unbounded` or `Closed` since precision doesn't matter in those.

This would make the call pass without changing the interface.

Reply:
I guess thinking about it more that should be fine. It makes sense that the function signatures be the same for the closed and open functions even if the precision does nothing on the closed ones. If we wanted viable defaults for precision we'd probably need to define the function for a bunch of different types which may not be worth it at this point.

Reply:
I would say that the default should be `eps` and for types where that value isn't defined it just needs to be passed in explicitly.